models.go

```diff
@@ -0,0 +1,145 @@
+package influxdb
+
+import (
+	"sort"
+	"time"
+
+	lp "github.com/influxdata/line-protocol"
+)
+
+// Metric is just a github.com/influxdata/line-protocol.Metric.
+// We alias here to keep abstractions from leaking.
+type Metric = lp.Metric
+
+// RowMetric is a Metric,
+// that has methods to make it easy to add tags and fields
+type RowMetric struct {
+	NameStr string
+	Tags    []*lp.Tag
+	Fields  []*lp.Field
+	TS      time.Time
+}
+
+// TagList returns a slice containing Tags of a Metric.
+func (m *RowMetric) TagList() []*lp.Tag {
```

Comment:
Might be worth attempting to prevent the user from directly modifying these items, by returning an interface.  This is something I've been meaning to do in Telegraf.

Reply:
I *really* don't like that.  Also it won't stop a user from doing it, they end up just using reflection if they really want to.

Reply:
It's more about hinting to the user that they shouldn't modify this without using the modification functions.  What don't you like about it?

Reply:
1) Generally returning interfaces is awkward.
2) needless virtualized function call.
3) needless extra function call.
4) Feels very java instead of go.  The only reason we even have a TagList method and helpers is so we can accept different types for encoding. 
3) It keeps the types clean, instead of having it return an interface that then has a method to return a tag list, it is clear what it is returning.

Reply:
What's a virtualized function call in Go?

Returning a concrete type could be nice but I think then it should be a copy of the data.  If this type is meant to allow users to use their own unlying implementation, it seems that returning a concrete type forces a copy in most cases.  Maybe this should be `[]Tag`?

Reply:
A virtualized function call means it has to do extra work on the method call, and it normally can't be inlined.
Internally it stores the pointers so it is only a copy of the pointer values not of the value the pointer points to.
I'm not sure what making it a copy gives us, besides the extra copying.